src/core/execute.c

```diff
@@ -5696,44 +5711,88 @@ int exec_spawn(Unit *unit,
-                                              "MESSAGE_ID=" SD_MESSAGE_SPAWN_FAILED_STR,
-                                              LOG_UNIT_INVOCATION_ID(unit),
-                                              LOG_UNIT_MESSAGE(unit, "Failed at step %s spawning %s: %m",
-                                                               status, command->path),
-                                              "EXECUTABLE=%s", command->path);
-                }
+        f = take_fdopen(&serialization_fd, "w+");
+        if (!f)
+                return log_unit_error_errno(unit, serialization_fd, "Failed to open serialization stream: %m");
+
+        fdset = fdset_new();
+        if (!fdset)
+                return log_oom();
 
-                _exit(exit_status);
+        r = unit_serialize(unit, f, fdset, /* switching_root */ false);
+        if (r < 0)
+                return log_unit_error_errno(unit, r, "Failed to serialize unit '%s': %m", unit->id);
+
+        r = exec_command_serialize(command, f);
+        if (r < 0)
+                return log_unit_error_errno(unit, r, "Failed to serialize ExecCommand: %m");
+
+        r = exec_parameters_serialize(params, f, fdset);
+        if (r < 0)
+                return log_unit_error_errno(unit, r, "Failed to serialize ExecParameters: %m");
+
+        r = exec_runtime_serialize(runtime, f, fdset);
+        if (r < 0)
+                return log_unit_error_errno(unit, r, "Failed to serialize ExecRuntime: %m");
+
+        r = exec_context_serialize(context, f, fdset);
+        if (r < 0)
+                return log_unit_error_errno(unit, r, "Failed to serialize ExecContext: %m");
+
+        r = exec_cgroup_context_serialize(cgroup_context, f);
+        if (r < 0)
+                return log_unit_error_errno(unit, r, "Failed to serialize CGroupContext: %m");
+
+        if (fseeko(f, 0, SEEK_SET) == (off_t) -1)
+                return log_unit_error_errno(unit, errno, "Failed to reseek on serialization stream: %m");
+
+        r = fd_cloexec(fileno(f), false);
+        if (r < 0)
+                return log_unit_error_errno(unit, errno, "Failed to set O_CLOEXEC on serialization fd: %m");
+
+        r = fdset_cloexec(fdset, false);
+        if (r < 0)
+                return log_unit_error_errno(unit, errno, "Failed to set O_CLOEXEC on serialized fds: %m");
+
+        /* The executor binary is pinned, to avoid compatibility problems during upgrades. */
+        xsprintf(cb_arg.serialization_fd_number, "%i", fileno(f));
+        cb_arg.executor_fd = unit->manager->executor_fd;
+        cb_arg.executor_path = FORMAT_PROC_FD_PATH(unit->manager->executor_fd);
+
+        if (stack_size == 0) {
+                struct rlimit rl;
+
+                r = getrlimit(RLIMIT_STACK, &rl);
+                if (r < 0)
+                        return log_unit_error_errno(unit, errno, "Failed to get RLIMIT_STACK: %m");
+
+                stack_size = MIN(rl.rlim_cur, page_size());
```

Comment:
hmm, pobably should round up to next page_size(), i..e ROUND_UP() should come handy there:

#27902